api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -385,8 +387,8 @@ public boolean getContactEmailTaken(String contactEmail) {
   /**
    *  Syncs the current user's training status from Moodle.
    */
-  public void syncComplianceTrainingStatus() throws org.pmiops.workbench.moodle.ApiException, NotFoundException {
-    syncComplianceTrainingStatus(userProvider.get());
+  public User syncComplianceTrainingStatus() throws org.pmiops.workbench.moodle.ApiException, NotFoundException {
+    return syncComplianceTrainingStatus(userProvider.get());
```

Comment:
why change this to return a user?

Reply:
This makes it easier to compare the user state before & after doing the status sync.

While we tend to make changes directly on the user object passed into these methods, it's theoretically possible to get a new User reference if the saveUserWithRetries process kicks in and fetches a new user from the database. So I thought making these methods explicitly return a User would be safest.

(Possible follow-up: change these methods to not modify the inbound User object and always return a new copy. I didn't bite this off in this PR since there isn't an easily-accessible User.clone() method... it may be this pattern would be too much effort to implement given the limited gains.)